gapic-generator/lib/gapic/presenters/gem_presenter.rb

```diff
@@ -195,6 +195,35 @@ def generate_rest_clients?
         @api.generate_rest_clients?
       end
 
+      ##
+      # @return [Boolean]
+      #
+      def generate_grpc_clients?
+        @api.generate_grpc_clients?
+      end
+
+      ##
+      # @return [:grpc] If gRPC is the default transport
+      # @return [:rest] if REST is the default transport
+      #
+      def default_transport
+        @api.default_transport
+      end
+
+      ##
+      # @return [String] Pretty name of the default transport
+      #
+      def default_transport_name
+        @api.default_transport == :grpc ? "gRPC" : "REST"
+      end
+
+      ##
+      # @return [Boolean]
+      #
+      def supports_multiple_transports?
+        @api.generate_rest_clients? && @api.generate_grpc_clients?
```

Comment:
I wonder if this will silently cause issues if transport=rest is set on a packages with no rest bindings for the services.
Generator also looks at the package: `if @api.generate_rest_clients? && package.first_service_with_rest`
https://github.com/googleapis/gapic-generator-ruby/blob/7e8d78c75bc49415ba11a09d4c5ab91dd8fd7e97/gapic-generator/lib/gapic/generators/default_generator.rb#L58

Maybe it's the other way around and we should throw if `generate_rest_clients?` is specified for a proto with no bindings.

Reply:
Yeah, interesting problem. I added a new commit that tries to address this by incorporating the presence of at least one method with rest bindings, into the (presenter-level) `generate_rest_clients?` logic itself. So `ServicePresenter#generate_rest_clients?` returns true only if REST clients are requested AND at least one of the service methods has REST bindings. Then `PackagePresenter#generate_rest_clients?` returns true only if at least one of its services returns true, and the gem itself returns true only if at least one of its packages returns true. Now each template should do the right thing depending on what it's concerned with. For example, the gem itself will declare that it includes a REST client overall (e.g. for documentation purposes) if there is _some_ REST code, even if not all services support REST. But some of its services may include REST and others may omit REST (e.g. for purposes of the constructors and service-level documentation).

Note: I also removed the `grpc_client_designation` logic because it was already inconsistent: it was changing for gRPC but not for REST. Now, we always just explicitly say `gRPC` or `REST` regardless of whether both are present or not.